docs/topic-guides/training-units.txt

```diff
@@ -0,0 +1,180 @@
+.. _topic-guides_training-units:
+
+################
+ Training Units
+################
+
+With the flexibility of Determined’s experiment configuration, the combinations of possible
+configuration settings available are numerous. This guide aims to help you better understand the
+options available for training units to make the right choice for your model.
+
+Determined supports configuring the following fields:
+
+-  ``searcher.max_length`` in most searchers
+
+-  ``searcher.length_per_round`` in PBT
+
+-  ``min_validation_period``
+
+-  ``min_checkpoint_period``
+
+-  ``searcher.budget`` when using ``adaptive_simple``
+
+in the following units:
+
+-  ``records``: A *record* is a single labeled example (sometimes called a sample).
+
+-  ``batches``: A *batch* is a group of records (the number of records in a batch is configured via the ``global_batch_size`` hyperparameter).
+
+-  ``epochs``: An *epoch* is typically single copy of the entire training data set.
+
+
+The sections below present different configurations that result in the same effective training
+length and behavior but use different training units.
+
+In general, conversions like the conversions between the configurations below will result
+in identical behavior except in the following cases:
+
+-  When a quantity is too granular to be converted to a less granular unit. For example, since
+   partial values aren't accepted, moving from 50 ``records`` and to ``epochs`` with 100 records per
+   epoch isn't possible. 
+
+-  When using a variable hyperparameter ``global_batch_size``. These configurations cannot be
+   expressed correctly in terms of batches, since expressing the training length in terms of batches
+   will result in a variable number of records.
+
+-  When using an adaptive searcher such as ``adaptive_simple``. When the length of each rung is
+   calculated, it is rounded up to the nearest whole unit. For example, an ASHA subsearch with 4
+   rungs, a max length of 1 epoch, 64 records per epoch and a global batch size of 1 will result in
+   trials in the final rung that run for 4 epochs, running one epoch in each rung. If the subsearch
+   is instead configured in terms of records (with a max length of 64 records), the trials in the
+   final rung would run for 1 epoch total, running 1, 4, 16 and 64 records by the end of each rung.
+
+To verify your search is working as intended before committing to a full run, you can use the
+preview search feature with the CLI command shown below.
+
+.. code::
+
+  det preview-search <configuration.yaml>
+
+*********
+ Batches
+*********
+
+If you know your model normally takes 900 batches to converge, in Determined you can specify the
+``max_length`` in terms of batches. The snippet below describes a searcher that
+trains a single trial on 900 batches, where a batch is defined by your model's dataloader.
```

Comment:
A batch would be defined by `global_batch_size` more so than the data loader, no?

Reply:
Yep, meant to remove this.